streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryTimeOrderedKeyValueBuffer.java

```diff
@@ -189,12 +199,13 @@ public boolean isOpen() {
 
     @Override
     public void close() {
+        open = false;
         index.clear();
         sortedMap.clear();
         dirtyKeys.clear();
         memBufferSize = 0;
         minTimestamp = Long.MAX_VALUE;
-        open = false;
```

Comment:
Why moving this?

Reply:
so that the buffer will report as "closed" immediately after this method is called, rather than only after the close operation has completed.

In retrospect, it seemed safer for :
* `open()` to set `open=true` at when opening has *completed*
* `close()` to set `open=false` when closing has *started*

This way, observers who first check that a store is open before querying would never be fooled into querying a partially closed store.